src/SearchParameters/index.js

```diff
@@ -365,9 +365,16 @@ SearchParameters.prototype = {
    * @return {SearchParameters}
    */
   addFacetRefinement : function addFacetRefinement( facet, value ) {
+    if( this.isRefined( facet, value ) ) {
+      return this;
```

Comment:
In the case isRefined( facet, value ) returns true, it would mean that there would be no changes, so might as well avoid creating a new instance (there are thousands of case we create useless new objects, true :D, it's just continuous improvement here)